packages/controls/src/form/TForm/TForm.js

```diff
@@ -60,7 +68,7 @@ function RenderForm(args) {
 					<Message.Header>{!isEmpty(warningArray) ? 'Some fields are not complete:' : `${errorHeader}:`}</Message.Header>
 					{/* Put it in a segment to make sure the errorMessage isn't to big when there are a lot of errors */}
 					<Segment style={{overflow: 'auto', maxHeight: 100}}>
-						{(!isEmpty(warningArray) ? warningArray : errorArray).map(warning => <div key={warning.message}>{warning.message}</div>)}
+						{(!isEmpty(warningArray) ? warningArray : errorArray).map((warning, index) => <div key={warning.message.concat(index)}>{warning.message}</div>)}
```

Comment:
It's usually not a good idea to use the index as the React key. There are a lot of articles out there explaining why.

Originally I was just using the warning text as the key. Can that not be used? Do we have duplicate warnings?

By the way, this is what's causing the `yarn test` command to fail. Run `yarn test` (in the packages/controls folder) to see why the test is failing.

Reply:
we do have warning messages that are the same so I can't use those...

I didn't want to fix the yarn test till everything is good to go...